### PR TITLE
Feature to change email and phone

### LINKED ISSUE #198: Allow email address and phone number to be changed
Currently, to edit email and password, one needs to send an email to an admin. Visit, https://zubhub.unstructured.studio/edit-profile and see email and password field. For reference, see screenshot below:

Screen Shot 2022-05-16 at 5 09 11 PM

Ideally, user should be able to change the address and phone number themselves without having to reach out to an admin.

### PR DESCRIPTION
**Pull request by iamhks**
## Summary

Allow email address and phone number to be changed
Closes #198 

## Changes

- Few changes to serializers in backend to accept change of email & phone
- Changes in frontend to implement the feature (new password field and editable phone & email fields)
- Made use of the existing login function to verify the current user and make changes accordingly

## Things to note:
I was wondering how directly exposing the backend to accept changes would not create a security issue as anyone can hit the api, and I tried the same. So, that scenario is handled by authorization token which is unique for each user.

## Screenshots
image
image

Second commit was done to show the error message of existing email/phone like this:
image

### CODE REVIEW

**Comment by tuxology** [comment]
@kamthamc and @NdibeRaymond Would need your help in this for review 🙏

**Comment by iamhks** [comment]
@kamthamc @NdibeRaymond Did you get some time to review this PR?

File: zubhub_backend/zubhub/creators/serializers.py
Comment (tuxology) [inline zubhub_backend/zubhub/creators/serializers.py]: Remove comments

**Comment by tuxology** [comment]
@iamhks I am testing this on staging as well, so this will take time. @kamthamc Is reviewing this in parallel

File: zubhub_backend/zubhub/creators/serializers.py
Comment (iamhks) [reply zubhub_backend/zubhub/creators/serializers.py]: Done

**Comment by iamhks** [comment]
> @iamhks I am testing this on staging as well, so this will take time. @kamthamc Is reviewing this in parallel

Sure @tuxology. I read about why it's taking time here: https://unstructured.studio/blog/zubhub-staging-server/
Please take your time.

File: zubhub_frontend/zubhub/src/views/edit_profile/EditProfile.jsx
Comment (chaitanya-cloudknox) [inline zubhub_frontend/zubhub/src/views/edit_profile/EditProfile.jsx:407]: I think the logic seems incorrect?
We need to show error if the status object has valid password property or the errors has valid password

Assuming status is the servers response

File: zubhub_frontend/zubhub/src/views/edit_profile/editProfileScripts.js
Comment (chaitanya-cloudknox) [inline zubhub_frontend/zubhub/src/views/edit_profile/editProfileScripts.js]: remove console.log

Comment (chaitanya-cloudknox) [reply zubhub_frontend/zubhub/src/views/edit_profile/editProfileScripts.js]: And there might be parsing error if error.message is not JSON. try to surround the block with try catch

Comment (chaitanya-cloudknox) [reply zubhub_frontend/zubhub/src/views/edit_profile/editProfileScripts.js]: FYI you can use async and awaits to fix the callbacks and doing password_match checks

File: zubhub_frontend/zubhub/src/views/edit_profile/EditProfile.jsx
Comment (iamhks) [reply zubhub_frontend/zubhub/src/views/edit_profile/EditProfile.jsx:407]: @chaitanya-cloudknox  As far as I understand we are using the helper text over here either to show that user has not entered a password or the password length is too short similar to our sign-up page.
The response from the api whether password is valid and changes have been pushed or not is shown via a toast message (screenshots attached in the commit message)
Am I missing something?

File: zubhub_frontend/zubhub/src/views/edit_profile/editProfileScripts.js
Comment (iamhks) [reply zubhub_frontend/zubhub/src/views/edit_profile/editProfileScripts.js]: @chaitanya-cloudknox Updated the PR for the first 2 comments. I actually tried using async and await only initially but it was always giving me undefined saying promise unresolved so had to use .finally and implement this way.

**Comment by tuxology** [comment]
@chaitanya-cloudknox are we good here? Please approve this once its good.  I'll begin tests. No more inputs from me.

**Comment by iamhks** [comment]
@chaitanya-cloudknox @kamthamc Do you have anything else to add? We are waiting for your approval

**Comment by iamhks** [comment]
@chaitanya-cloudknox @kamthamc Gentle follow-up, waiting for your approval

File: zubhub_frontend/zubhub/src/views/edit_profile/EditProfile.jsx
Comment (tuxology) [reply zubhub_frontend/zubhub/src/views/edit_profile/EditProfile.jsx:407]: @chaitanya-cloudknox Yes, the issue for now is that we don't have a consistent way to show error, we are just showing the toast sometimes and sometimes we are just adding a small red text below the field. I feel this is fine for now.

**Review by tuxology — APPROVED** [review]

**Pull request merged**
